src/types.ts

```diff
@@ -217,6 +217,7 @@ export interface PhaseConfig<
     ) => boolean | void | { next: string };
     onBegin?: (state: State<G, CtxWithPlugins>) => any;
     onEnd?: (state: State<G, CtxWithPlugins>) => any;
+    next?: (state: State<G, CtxWithPlugins>) => any;
```

Comment:
Same here, we can specify the return type of `next`:

```suggestion
    next?: (state: State<G, CtxWithPlugins>) => string | void;
```